src/vm/methodtable.cpp

```diff
@@ -3029,23 +3087,31 @@ void  MethodTable::AssignClassifiedEightByteTypes(SystemVStructRegisterPassingHe
         }
 
         // Calculate the eightbytes and their types.
-        unsigned int accumulatedSizeForEightByte = 0;
-        unsigned int currentEightByteOffset = 0;
-        unsigned int currentEightByte = 0;
 
         int lastFieldOrdinal = sortedFieldOrder[largestFieldOffset];
         unsigned int offsetAfterLastFieldByte = largestFieldOffset + helperPtr->fieldSizes[lastFieldOrdinal];
         SystemVClassificationType lastFieldClassification = helperPtr->fieldClassifications[lastFieldOrdinal];
 
-        unsigned offset = 0;
-        for (unsigned fieldSize = 0; offset < helperPtr->structSize; offset += fieldSize)
+        unsigned int usedEightBytes = 0;
+        unsigned int accumulatedSizeForEightBytes = 0;
+        bool foundFieldInEightByte = false;
+        for (unsigned int offset = 0; offset < helperPtr->structSize; offset++)
         {
             SystemVClassificationType fieldClassificationType;
+            unsigned fieldSize = 0;
```

Comment:
Please use `unsigned int` or `uint32_t`, not a plain `unsigned`.

Reply:
What's wrong with `unsigned` ?

Reply:
Technically nothing, but it is generally considered an antiquated way of declaring an `unsigned int` and in this PR is inconsistent since all other cases use `unsigned int`. I also personally prefer the use of `uint32_t` because it tells me the bitness without me thinking about target platform. In this case I don't think we compile on any platform where `int` is 16, but I tend to use it for clarity.

It is definitely a personal preference. If this is still considered a fine practice, then ignore.

Reply:
+1 I think not everyone knows what just unsigned means.

Reply:
Maybe this is going to be a religious argument, but I think any C/C++ programmer should know what plain `unsigned` means. We have almost interest in compiling with a toolset that doesn't think `int` or `unsigned int` isn't 32-bits, on any platform. (I think `wchar_t` is the only case where it differs in size by platform, and it's a huge pain.) Seems unlikely we would change all our code to use the size-specific types (e.g., `uint32_t`) instead of `unsigned` or `unsigned int`, etc.